src/number_theory/bernoulli.lean

```diff
@@ -139,8 +164,48 @@ begin
   rw ← cast_sub hk,
   congr',
   field_simp [show ((n - k : ℕ) : ℚ) + 1 ≠ 0, by {norm_cast, simp}],
-  -- down to nat
   norm_cast,
   rw [mul_comm, nat.sub_add_eq_add_sub hk],
   exact choose_mul_succ_eq n k,
 end
+
+open power_series
+open nat
+
+@[simp] lemma constant_coeff_exp : constant_coeff ℚ (exp ℚ) = 1 := rfl
+
+theorem algebra_map_id (q : ℚ) : algebra_map ℚ ℚ q = q :=
+begin
+  rw [show algebra_map ℚ ℚ = (by refine_struct { to_fun := id}; tidy),
+    by simp only [eq_iff_true_of_subsingleton]], refl,
+end
```

Comment:
```suggestion
rat.cast_id _
```
Although this lemma should probably move to `rat.algebra_map_self` or similar

Reply:
Perhaps best as
```
@[simp] theorem rat.rat_algebra_map_rat : algebra_map ℚ ℚ = ring_hom.id _ :=
ring_hom.ext rat.cast_id
```

Reply:
Done

Reply:
The lemma belongs in a different file - I span off #5970 to avoid you having to rebuild everything in this PR

Reply:
That PR is now merged - you should be able to remove this lemma after merging master

Reply:
Apologies, somehow I missed this message. I have removed the lemma. Thank you!